h2o-genmodel/src/main/java/hex/genmodel/algos/tree/Graph.java

```diff
@@ -0,0 +1,54 @@
+package hex.genmodel.algos.tree;
+
+import java.io.PrintStream;
+import java.util.ArrayList;
+
+/**
+ * Graph for representing a forest.
+ */
+public class Graph {
```

Comment:
Suggestion: rename into `SharedTreeGraph` or similar, in order to avoid future potential name clashes with other types of graphs. Same applies to `Node` and `Subgraph`.